console/services/market_app_service.py

```diff
@@ -908,66 +908,81 @@ def _patch_rainbond_app_tag(self, eid, apps):
     def _get_rainbond_app_min_memory(self, apps_model_versions):
         apps_min_memory = dict()
         for app_model_version in apps_model_versions:
-            if not apps_min_memory.get(app_model_version.app_id):
-                min_memory = 0
-                try:
-                    app_temp = json.loads(app_model_version.app_template)
-                    for app in app_temp.get("apps"):
-                        if app.get("extend_method_map"):
-                            try:
-                                if app.get("extend_method_map").get("init_memory"):
-                                    min_memory += int(app.get("extend_method_map").get("init_memory"))
-                                else:
-                                    min_memory += int(app.get("extend_method_map").get("min_memory"))
-                            except Exception:
-                                pass
-                    apps_min_memory[app_model_version.app_id] = min_memory
-                except ValueError:
-                    apps_min_memory[app_model_version.app_id] = min_memory
+            min_memory = 0
+            try:
+                app_temp = json.loads(app_model_version.app_template)
+                for app in app_temp.get("apps"):
+                    if app.get("extend_method_map"):
+                        try:
+                            if app.get("extend_method_map").get("init_memory"):
+                                min_memory += int(app.get("extend_method_map").get("init_memory"))
+                            else:
+                                min_memory += int(app.get("extend_method_map").get("min_memory"))
+                        except Exception:
+                            pass
+                apps_min_memory[app_model_version.app_id] = min_memory
+            except ValueError:
+                pass
+            if min_memory <= apps_min_memory.get(app_model_version.app_id, 0):
+                apps_min_memory[app_model_version.app_id] = min_memory
         return apps_min_memory
 
     # patch rainbond app versions
-    def _patch_rainbond_app_versions(self, eid, apps, is_complete=None):
+    def _patch_rainbond_app_versions(self, eid, apps, is_complete):
         app_ids = [app.app_id for app in apps]
         versions = rainbond_app_repo.get_rainbond_app_version_by_app_ids(eid, app_ids, is_complete)
```

Comment:
It is recommended to query only fields other than the template. speed up queries.